deployment/terraform/coordinator.go

```diff
@@ -185,23 +169,15 @@ func (t *Terraform) StopCoordinator() (coordinator.Status, error) {
 
 // GetCoordinatorStatus returns information about the status of the
 // coordinator in the current load-test deployment.
-func (t *Terraform) GetCoordinatorStatus() (coordinator.Status, error) {
+func GetCoordinatorStatus(dpConfig *deployment.Config, output *Output) (coordinator.Status, error) {
 	var status coordinator.Status
-	if err := t.preFlightCheck(); err != nil {
-		return status, err
-	}
-
-	output, err := t.Output()
```

Comment:
The refactoring out of `PreFlightCheck` makes total sense to me.

I am wondering whether it's possible to maintain the output caching logic inside the terraform struct itself? It seems a bit odd to move this responsibility to consumer packages. I am considering alternatives like storing the output struct within the Terraform struct itself, and then either have a separate method called `CachedOutput` or pass a flag or something to the `Output` method to return the cached or a new response.

That way we still keep the logic of caching/not caching within the terraform package itself.

Reply:
I think that's a great idea. It would mean reverting the coordinator functions to be methods of `terraform.Terraform` but at least we'd avoid the slightly awkward parameters. I like it :+1: